Convert zaphod-config to a module.

amacleod opened the pull request:
Adds a `zephyr/module.yml` file to zaphod-config so that other zmk-config repos can reference it as a Zephyr module.

amacleod committed:
Trying an extremely simple module YAML to see what happens.

amacleod committed:
Adding build settings to module.yml

Much like a hitchhiker and their towel, a module with boards or
shields in it needs to know where its board_root is.

amacleod committed:
Adding a little README showing how to use the config as a module.

amacleod committed:
Merge pull request #1 from amacleod/zaphod-module

Merge Zaphod config module-ification to `main` on this fork.

amacleod commented:
If the README is extraneous, feel free to drop it

petejohanson commented on zephyr/module.yml:
```diff
@@ -0,0 +1,5 @@
+---
+name: zephyr-config
```
This should be something better matching, e.g. 


```suggestion
name: zaphod-config
```

amacleod replied:
Oh my goodness! I must have gone a little snow-blind with all the Zs and PHs :)

petejohanson reviewed (changes requested):
Thanks for the PR! Just one nitpick comment.

amacleod committed:
Fix thinko: zephyr-config to zaphod-config

I definitely meant for it to be zaphod-config in the first place.

Co-authored-by: Pete Johanson <peter@peterjohanson.com>

amacleod committed:
Fixing Zaphod Bongo Cat for LVGL 8.3 and Zephyr 3.5.0, and to work as a module. (#2)

* Copying ZAPHOD_BONGO_CAT config declaration into Kconfig.shield, duplicating what is in plain Kconfig.

* Updating include directories for Zephyr stuff: kernel.h and logging/log.h

* Trying to build things up one step at a time.

* Fix module name.

* Selecting LV_USE_IMG when ZMK_DISPLAY is enabled on zaphod_lite.

* In newer versions of Zephyr, the logging header is under a zephyr directory.

* Change LVGL to LV in config selections.

* Update Bongo Cat widget for LVGL 8.3

- there is no longer a `lv_anim_value_t` typedef, but `int32_t` works
  for indexing images in animator functions.
- `lv_image_create` only takes one parameter now: the parent object.
- `lv_img_set_auto_size` no longer exists.

* Minor whitespace cleanup.

* Setting width and height to LV_SIZE_CONTENT for auto-sizing.

I think the object might already be auto-sized by default, but it
can't hurt to be explicit about it.

* Removing LV_USE_IMG from defconfig since it is in the main Kconfig

Images and animation are only needed by this shield if the Bongo Cat
WPM widget is enabled. That selection is handled in
boards/shields/zaphod_lite/Kconfig itself.

* Reverting Kconfig.shield changes.

The configurations in plain Kconfig ought to be sufficient.

* Revert "Reverting Kconfig.shield changes."

This reverts commit 9c7761dd06c9b05b76e8b08ed9ddc75ca6f4c90e.

* Repeating LVGL -> LV correction in Kconfig.shield

amacleod commented:
> Thanks for the PR! Just one nitpick comment.

Thanks! I fixed the module name, and also made some code modifications to get things compiling with Zephyr 3.5.0 and LVGL 8.3. If it would be better to split "module stuff" and "widget compatibility stuff" into separate PRs, let me know. I think it shouldn't be too hard to tease the two apart if needed.

petejohanson commented:
Can we fix up the GHA build file to use the standard one while we are at it?

amacleod commented:
> Can we fix up the GHA build file to use the standard one while we are at it?

I will take a look! It should just use the same one from `zmkfirmware/zmk` like you get when you make a new personal config, right?

amacleod committed:
Use zmkfirmware/zmk standard GitHub Actions workflow.

amacleod commented:
One thing I do not fully understand is why copying the `ZAPHOD_BONGO_CAT` stuff from `Kconfig` to `Kconfig.shield` helped. I have the vague notion that one of them is used when building directly and the other is used when included as a module.  What are the rules for which Kconfig is used in what circumstances?

amacleod committed:
fix: Update bongo cat widget for LVGL 9

- lv_img_dsc_t → lv_image_dsc_t
- LV_IMG_CF_INDEXED_1BIT → LV_COLOR_FORMAT_I1
- lv_img_create → lv_image_create
- lv_img_set_src → lv_image_set_src
- Remove lv_image_header_t.always_zero (removed in LVGL 9)
